patches/tModLoader/Terraria/Recipe.TML.cs

```diff
@@ -62,16 +66,277 @@ public sealed class Condition : ICondition
+			mod ??= this.Mod;
+			
+			int type = mod.ItemType(itemName);
+
+			if (type == 0)
+				throw new RecipeException($"The item {itemName} does not exist in the mod {mod.Name}.\r\nIf you are trying to use a vanilla item, try removing the first argument.");
+
+			return AddIngredient(type, stack);
+		}
+
+		/// <summary>
+		/// Adds an ingredient to this recipe of the given type of item and stack size.
+		/// </summary>
+		/// <param name="item">The item.</param>
+		/// <param name="stack">The stack.</param>
+		public Recipe AddIngredient(ModItem item, int stack = 1) => AddIngredient(item.item.type, stack);
+
+		/// <summary>
+		/// Adds an ingredient to this recipe of the given type of item and stack size.
+		/// </summary>
+		/// <param name="item">The item.</param>
+		/// <param name="stack">The stack.</param>
+		public Recipe AddIngredient<T>(int stack = 1) where T : ModItem
+			=> AddIngredient(ModContent.ItemType<T>(), stack);
+
+		/// <summary>
+		/// Adds a recipe group ingredient to this recipe with the given RecipeGroup name and stack size. Vanilla recipe groups consist of "Wood", "IronBar", "PresurePlate", "Sand", and "Fragment".
+		/// </summary>
+		/// <param name="name">The name.</param>
+		/// <param name="stack">The stack.</param>
+		/// <exception cref="RecipeException">A recipe group with the name " + name + " does not exist.</exception>
+		public Recipe AddRecipeGroup(string name, int stack = 1) {
+			if (!RecipeGroup.recipeGroupIDs.ContainsKey(name))
+				throw new RecipeException($"A recipe group with the name {name} does not exist.");
+
+			int id = RecipeGroup.recipeGroupIDs[name];
+			var group = RecipeGroup.recipeGroups[id];
+			
+			AddIngredient(group.IconicItemId, stack);
+			RequireGroup(name);
+
+			return this;
+		}
+
+		/// <summary>
+		/// Adds a recipe group ingredient to this recipe with the given RecipeGroupID and stack size. Vanilla recipe group IDs can be found in Terraria.ID.RecipeGroupID and modded recipe group IDs will be returned from RecipeGroup.RegisterGroup.
+		/// </summary>
+		/// <param name="recipeGroupId">The RecipeGroupID.</param>
+		/// <param name="stack">The stack.</param>
+		/// <exception cref="RecipeException">A recipe group with the ID " + recipeGroupID + " does not exist.</exception>
+		public Recipe AddRecipeGroup(int recipeGroupId, int stack = 1) {
+			if (!RecipeGroup.recipeGroups.ContainsKey(recipeGroupId))
+				throw new RecipeException($"A recipe group with the ID {recipeGroupId} does not exist.");
+			
+			RecipeGroup rec = RecipeGroup.recipeGroups[recipeGroupId];
+			
+			AddIngredient(rec.IconicItemId, stack);
+			RequireGroup(recipeGroupId);
+
+			return this;
+		}
+
+		/// <summary>
+		/// Adds a recipe group ingredient to this recipe with the given RecipeGroup.
+		/// </summary>
+		/// <param name="recipeGroup">The RecipeGroup.</param>
+		public Recipe AddRecipeGroup(RecipeGroup recipeGroup, int stack = 1) {
```

Comment:
I see this parameter version slipped in here, but is it necessary? Is there a use case for it?